packaging/common/substitute.pl

```diff
@@ -56,21 +53,163 @@ sub fix_pkg_rev {
     # strip pkg_type and maybe a zero, else assign pkg_rev = 1
     $pkg_rev = $pkg_rev =~ s/$type_match_str// || 1;
     return $pkg_rev;
+}
+
+sub find_platform_info {
```

Comment:
I could comment this a bit better... but suffice it to say :

- check for the contents or output of "lsb_release" [which tracks well with unique names]
- check for the contents of variable-holding files in /etc
- check for the existence of distinct file names themselves and their contents too